feder/institutions/serializers.py

```diff
@@ -56,6 +59,31 @@ def update(self, instance, validated_data):
             instance.parents.set(validated_data["parents"])
         return super().update(instance, validated_data)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._jst_cache = {}
+
+    def _get_jst(self, jst_id):
+        """
+        Gets JST instance by given id and caches the results to avoid further
+        unnecessary queries.
+        """
+        jst = self._jst_cache.get(jst_id)
```

Comment:
Drop that cache, please. Use `select_related` instead.

Reply:
Yes, if there are max 2 levels of JST parents, then it's doable. I will change that.

Reply:
Such an assumption results from the Polish Constitution.